lib/utils/log/slog_text_handler.go

```diff
@@ -362,3 +367,21 @@ func (s *SlogTextHandler) WithGroup(name string) slog.Handler {
 	s2.groups = append(s2.groups, name)
 	return s2
 }
+
+type SlogTextHandlerOutputIO struct {
+	mu  sync.Mutex
+	out io.Writer
+}
+
+func NewSlogTextHandlerOutputIO(w io.Writer) *SlogTextHandlerOutputIO {
+	return &SlogTextHandlerOutputIO{out: w}
+}
+
+func (o *SlogTextHandlerOutputIO) Write(buf *buffer, rawComponent string, level slog.Level) error {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+
+	// TODO: Should I do buf.WriteByte instead of append here?
+	_, err := o.out.Write(append(*buf, '\n'))
```

Comment:
We used to always add `\n`, but now it's needed only for io.Writer. I suspect `append` might not be optimal?

I could call `buf.WriteByte`, which would work for now. It just felt wrong to me to mutate a buffer passed from outside which in theory can be passed elsewhere too (e.g., if we ever end up supporting multiple outputs).

Reply:
It looks like os_log automatically handles trailing newline, that is the newline isn't shown when you read logs with `log stream`/`log show`/Console.app. So I brought back the original handling of newline to not mess up the behavior around `bufPool` (which I don't intuitively understand).